modules/dxp/apps/app-builder-workflow/app-builder-workflow-web/src/main/resources/META-INF/resources/js/pages/entry/ViewEntry.es.js

```diff
@@ -37,11 +40,28 @@ export default function ViewEntry({
 		appWorkflowDefinitionId,
 		appWorkflowTasks,
 	} = useAppWorkflow(appId);
-	const {
-		dataDefinition,
-		dataLayout: {dataLayoutPages},
-		isLoading,
-	} = useDataLayout(dataLayoutId, dataDefinitionId);
+
+	const dataLayoutIds = useMemo(() => {
+		return appWorkflowTasks?.reduce(
+			(dataLayoutIds, {appWorkflowDataLayoutLinks}) => [
+				...dataLayoutIds,
+				...appWorkflowDataLayoutLinks?.reduce(
+					(stepDataLayoutIds, {dataLayoutId}) =>
+						dataLayoutIds.includes(dataLayoutId)
+							? stepDataLayoutIds
+							: [...stepDataLayoutIds, dataLayoutId],
+					[]
+				),
+			],
+			[Number(dataLayoutId)]
+		);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [appWorkflowTasks]);
+
+	const {dataDefinition, dataLayouts, isLoading} = useDataLayouts(
+		dataLayoutIds,
+		dataDefinitionId
```

Comment:
@naraandrade you should reorder the arguments, as you reordered at function definition.